cpp/src/search/search.cu

```diff
@@ -43,33 +45,52 @@
 
 namespace cudf {
 namespace {
-template <typename DataIterator,
-          typename ValuesIterator,
-          typename OutputIterator,
-          typename Comparator>
-void launch_search(DataIterator it_data,
-                   ValuesIterator it_vals,
-                   size_type data_size,
+
+struct make_lhs_index {
+  __device__ lhs_index_type operator()(size_type i) const { return static_cast<lhs_index_type>(i); }
+};
+
+struct make_rhs_index {
+  __device__ rhs_index_type operator()(size_type i) const { return static_cast<rhs_index_type>(i); }
+};
+
+auto make_lhs_index_counting_iterator(size_type start)
+{
+  return cudf::detail::make_counting_transform_iterator(start, make_lhs_index{});
+};
+
+auto make_rhs_index_counting_iterator(size_type start)
+{
+  return cudf::detail::make_counting_transform_iterator(start, make_rhs_index{});
+};
```

Comment:
Let's move these utility functions into `strong_index.hpp` because they will be used a lot.

Reply:
In addition, please marked them as `inline`.

Reply:
I'm iffy about a separate header (are they useful outside of the row comparator?), but I do agree they should at least be in the row comparator header.

Reply:
Done in 73c4b27f7461d80bee1276532f0ed89a438b5866.